app/scripts/controllers/encryption-public-key.ts

```diff
@@ -100,6 +102,8 @@ export default class EncryptionPublicKeyController extends BaseControllerV2<
 
   private _keyringController: KeyringController;
 
+  private _coreKeyringController: CoreKeyringController;
+
```

Comment:
Seems like `EncryptionPublicKeyController` uses the keyring controller for two reasons:
- Obtaining the type of a keyring (which you already covered)
- Getting the encryption public key for an account (which we don't even support in the core KeyringController)
To fix the remaining issue I think we should add a `getEncryptionPublicKey` function on `coreKeyringController`, to expose the homonym EthKeyringController's function - this would be enough to drop the old keyringcontroller usage completely

I will leave a reflection on MMIController in another comment